src/app_service/service/stickers/service.nim

```diff
@@ -0,0 +1,237 @@
+      running: cast[ByteAddress](addr self.threadpool.running)
+    )
+    self.threadpool.start(arg)
+
+  proc setGasEstimate*(self: Service, estimateJson: string) {.slot.} =
+    let estimateResult = Json.decode(estimateJson, tuple[estimate: int, uuid: string])
+    self.events.emit(SIGNAL_STICKER_GAS_ESTIMATED, StickerGasEstimatedArgs(estimate: estimateResult.estimate, uuid: estimateResult.uuid))
+
+  # the [T] here is annoying but the QtObject template only allows for one type
+  # definition so we'll need to setup the type, task, and helper outside of body
+  # passed to `QtObject:`
+  proc estimate*(self: Service, packId: int, address: string, price: string, uuid: string) =
+    let arg = EstimateTaskArg(
+      tptr: cast[ByteAddress](estimateTask),
+      vptr: cast[ByteAddress](self.vptr),
+      slot: "setGasEstimate",
+      packId: packId,
+      address: address,
+      price: price,
+      uuid: uuid
+    )
+    self.threadpool.start(arg)
+
+  proc addStickerToRecent*(self: Service, sticker: StickerDto, save: bool = false) =
+    self.recentStickers.insert(sticker, 0)
+    self.recentStickers = self.recentStickers.deduplicate()
+    if self.recentStickers.len > 24:
+      self.recentStickers = self.recentStickers[0..23] # take top 24 most recent
+    if save:
+      status_stickers.saveRecentStickers(self.recentStickers.mapIt($it.hash))
+
+  proc getRecentStickers*(self: Service): seq[StickerDto] =
+    let stickers = status_stickers.getRecentStickers()
+
+    for sticker in stickers:
+      let stickerDto = StickerDto(
+        hash: sticker.hash,
+        packId: sticker.packId
+      )
+      self.addStickerToRecent(stickerDto)
+
+    result = self.recentStickers
+      
+  proc getNumInstalledStickerPacks*(self: Service): int =
+    return self.installedStickerPacks.len
+
+  proc installStickerPack*(self: Service, packId: int) =
+    if not self.availableStickerPacks.hasKey(packId):
+      return
+    let pack = self.availableStickerPacks[packId]
+    self.installedStickerPacks[packId] = pack
+
+    let json = %* {}
+    for packId, pack in self.installedStickerPacks.pairs:
+      json[$packId] = %(pack)
+    
+    status_stickers.saveInstalledStickerPacks($json)
+
+  proc uninstallStickerPack*(self: Service, packId: int) =
+    if not self.installedStickerPacks.hasKey(packId):
+      return
+    let pack = self.availableStickerPacks[packId]
+    self.installedStickerPacks.del(packId)
+
+    let json = %* {}
+    for packId, pack in self.installedStickerPacks.pairs:
+      json[$packId] = %(pack)
+
+    status_stickers.saveInstalledStickerPacks($json)
+
+  proc sendSticker*(self: Service, chatId: string, replyTo: string, sticker: StickerDto) =
+    let stickerToSend = Sticker(
+        hash: sticker.hash,
+        packId: sticker.packId
+      )
+    var response = status_chat.sendStickerMessage(chatId, replyTo, stickerToSend)
+    self.events.emit("stickerSent", StickerArgs(sticker: stickerToSend, save: true))
```

Comment:
@jrainville these emits here "stickerSent", "chatUpdate", "sendingMessage" are not relevant for the new (refactored) part of the app (not used anywhere).

Reply:
I removed two of them, because they were indeed useless.
The last one I had to keep to make it possible to test. 
I added a TODO to remove/change it once the new chat is fully refactored